testsuite/features/step_definitions/cobbler_steps.rb

```diff
@@ -250,3 +249,34 @@
     _out, _code = $server.run('cobbler sync')
   end
 end
+
+When(/^I start local monitoring of Cobbler$/) do
+  cobbler_conf_file = '/etc/cobbler/logging_config.conf'
+  cobbler_log_file = '/var/log/cobbler/cobbler_debug.log'
+  $server.run("rm #{cobbler_log_file}", check_errors: false)
+  _result, code = $server.run("test -f #{cobbler_conf_file}.old", check_errors: false)
+  if !code.zero?
+    handler_name = 'FileLogger02'
+    handler_class = "\"\n[handler_#{handler_name}]\n" \
+                  "class=FileHandler\n" \
+                  "level=DEBUG\n" \
+                  "formatter=Logfile\n" \
+                  "args=('#{cobbler_log_file}', 'a')\""
+    command = "cp #{cobbler_conf_file} #{cobbler_conf_file}.old && " \
+              "line_number=`awk \"/\\\[handlers\\\]/{ print NR; exit }\" #{cobbler_conf_file}` && " \
+              "sed -e \"$(($line_number + 1))s/$/,#{handler_name}/\" -i #{cobbler_conf_file} && " \
+              "line_number=`awk \"/\\\[logger_root\\\]/{ print NR; exit }\" #{cobbler_conf_file}` && " \
+              "sed -e \"$(($line_number + 2))s/$/,#{handler_name}/\" -i #{cobbler_conf_file} && " \
+              "echo -e #{handler_class} >> #{cobbler_conf_file}"
+    $server.run("#{command} && systemctl restart cobblerd", check_errors: false)
+  else
+    $server.run('systemctl restart cobblerd', check_errors: false)
+  end
+end
+
+Then(/^I check for Cobbler errors in the local logs$/) do
+  cobbler_log_file = '/var/log/cobbler/cobbler_debug.log'
+  output, code = $server.run("grep -i error #{cobbler_log_file}", check_errors: false)
```

Comment:
I'll make my suggestion more direct:

- Let's use https://ruby-doc.org/stdlib-3.0.2/libdoc/json/rdoc/JSON.html to load the log objects
- Let's use https://github.com/madzak/python-json-logger to log to disk in the JSON format

We gain from this the distinct opportunity to have a well-defined object that we can parse as expected. Parsing clear text with `grep` will never work as intended. My remarks earlier were precisely about the fact that with `grep -i error` you will catch messages that are not errors in the context of Cobbler because Cheetah is omitting messages about variables that haven't been found (which is to be expected in the test suite).

Reply:
I is also available on Leap 15.4 which the controller runs on.
```bash
suma-bv-43-ctl:~/spacewalk/testsuite # zypper se python-json-logger
Loading repository data...
Reading installed packages...

S | Name                       | Summary                                      | Type
--+----------------------------+----------------------------------------------+--------
  | python3-python-json-logger | A python library adding a json log formatter | package
```

Reply:
My concern was addressed and thus I resolve this discussion.

Reply:
It seems that `python3-python-json-logger` is available only from Package Hub on SLE :cry:

Reply:
I don't see the problem here as no customer is affected by this?